src/components/ScreenWrapper.tsx

```diff
@@ -0,0 +1,252 @@
+import {useNavigation} from '@react-navigation/native';
+import {StackNavigationProp} from '@react-navigation/stack';
+import React, {ForwardedRef, forwardRef, ReactNode, useEffect, useRef, useState} from 'react';
+import {DimensionValue, Keyboard, PanResponder, StyleProp, View, ViewStyle} from 'react-native';
+import {PickerAvoidingView} from 'react-native-picker-select';
+import {EdgeInsets} from 'react-native-safe-area-context';
+import useEnvironment from '@hooks/useEnvironment';
+import useInitialDimensions from '@hooks/useInitialWindowDimensions';
+import useKeyboardState from '@hooks/useKeyboardState';
+import useNetwork from '@hooks/useNetwork';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import * as Browser from '@libs/Browser';
+import {RootStackParamList} from '@libs/Navigation/types';
+import toggleTestToolsModal from '@userActions/TestTool';
+import CONST from '@src/CONST';
+import CustomDevMenu from './CustomDevMenu';
+import HeaderGap from './HeaderGap';
+import KeyboardAvoidingView from './KeyboardAvoidingView';
+import OfflineIndicator from './OfflineIndicator';
+import SafeAreaConsumer from './SafeAreaConsumer';
+import TestToolsModal from './TestToolsModal';
+
+type ChildrenProps = {
+    insets?: EdgeInsets;
+    safeAreaPaddingBottomStyle?: {
+        paddingBottom?: DimensionValue;
+    };
+    didScreenTransitionEnd: boolean;
+};
+
+type ScreenWrapperProps = {
+    /** Returns a function as a child to pass insets to or a node to render without insets */
+    children: ReactNode | React.FC<ChildrenProps>;
+
+    /** A unique ID to find the screen wrapper in tests */
+    testID: string;
+
+    /** Additional styles to add */
+    style?: StyleProp<ViewStyle>;
+
+    /** Additional styles for header gap */
+    headerGapStyles?: StyleProp<ViewStyle>;
+
+    /** Styles for the offline indicator */
+    offlineIndicatorStyle?: StyleProp<ViewStyle>;
+
+    /** Whether to include padding bottom */
+    includeSafeAreaPaddingBottom?: boolean;
+
+    /** Whether to include padding top */
+    includePaddingTop?: boolean;
+
+    /** Called when navigated Screen's transition is finished. It does not fire when user exit the page. */
+    onEntryTransitionEnd?: () => void;
+
+    /** The behavior to pass to the KeyboardAvoidingView, requires some trial and error depending on the layout/devices used.
+     *  Search 'switch(behavior)' in ./node_modules/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.js for more context */
+    keyboardAvoidingViewBehavior?: 'padding' | 'height' | 'position';
```

Comment:
if this is something that is reusable maybe extract it to some global type?

Reply:
It is used once in the entire app, I don't think it's worth it 🤷